controllers/loadtest_controller.go

```diff
@@ -305,7 +305,7 @@ func (r *LoadTestReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 				if updateErr := r.Status().Update(ctx, test); updateErr != nil {
 					logWithServer.Error(updateErr, "failed to update status after failure to construct a pod for server")
 				}
-				return ctrl.Result{}, err
+				return ctrl.Result{Requeue: false}, nil
```

Comment:
why we do not want to requeue the event due to this failure?

Reply:
I made the distinction between two types of errors in the controller.

- There are reconciliation errors, which occur when something unexpected went wrong with the reconcile. For example, a pod could not be created. This example is something that should be retried on another reconcile.

- There are user errors, which occur when something goes wrong due to what the user has done. For example, this code uses the PodBuilder type to construct a pod for a server. It failed due to a problem with the configuration of the test itself. In this situation and in other user errors, the reconciliation should not be retried. The controller successfully "reconciled", but it updates the status to show that something has went horribly wrong. 

It is a fair to consider a failure when updating the status for user errors. In that situation, it should probably be a reconciliation error that is returned to retry. In this code, however, I just ignore the failure to update the status and log it. Perhaps it can be an enhancement, though it complicates the flow a bit.

Reply:
Seems like this return associated to error possibly returned by PodforServer, the only error it could return is "could not determine pool for server %q (no explicit value or default", make scene not requeue this since it has nothing to do with the reconciliation process. :)